src/main/java/org/openpnp/machine/reference/feeder/ReferenceStripFeeder.java

```diff
@@ -223,11 +223,20 @@ public Location getPickLocation() throws Exception {
     public void feed(Nozzle nozzle) throws Exception {
         setFeedCount(getFeedCount() + 1);
 
+        // normally the feeder will automatically disable when it runs out but
+        // if the user re-enables and tries to feed, throw an exception that we are out of parts.
         if ((maxFeedCount > 0) && (feedCount > maxFeedCount)) {
 			throw new Exception("Tried to feed part: " + part.getId() + "  Feeder " + name + " empty.");
 		}
 
         updateVisionOffsets(nozzle);
+        
+        // automatically disable when we reach the last part, openpnp will try
```

Comment:
Hi @desertgreg - with my recent fixes to the job processor this is handled automatically. The feeder should just throw an exception if it can't feed. Is there a need for this additional change?

Reply:
Upon further thought, if I only throw the exception, will a message box pop up and wait for user input?  Thats what was happening for me.  With this change the system seamlessly continues placing parts until the last feeder runs out.  So it might be better to keep this change but I will defer to your opinion :-)

Reply:
Hi Jason, what do you think?  I tested it again without my auto-disabling code and I think the user experience is worse.  Without this code, the job pauses and puts a message box up that the user has to click through and then they have to press 'play' on the job again to get it going.  With this code in, there is no pause or user interaction needed, openpnp just moves to the next feeder.  I suppose you could implement this on your side though.  You could only show the exception message if you can't find another feeder for the given part.

Reply:
@desertgreg You need to set the pick retry count for the part in the Parts panel. This activates the new setting. The reason it should be done this way is so that it acts the same for all feeders - and this leaves it up to the user as to whether they want to auto advance to a second feeder.